src/main/java/uk/gov/ons/ctp/response/casesvc/service/CaseIACService.java

```diff
@@ -1,18 +1,92 @@
 package uk.gov.ons.ctp.response.casesvc.service;
 
+import com.godaddy.logging.Logger;
+import com.godaddy.logging.LoggerFactory;
+import java.sql.Timestamp;
+import java.time.Clock;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
 import uk.gov.ons.ctp.common.error.CTPException;
-import uk.gov.ons.ctp.common.service.CTPService;
+import uk.gov.ons.ctp.response.casesvc.client.InternetAccessCodeSvcClient;
+import uk.gov.ons.ctp.response.casesvc.domain.model.CaseEvent;
 import uk.gov.ons.ctp.response.casesvc.domain.model.CaseIacAudit;
+import uk.gov.ons.ctp.response.casesvc.domain.repository.CaseEventRepository;
+import uk.gov.ons.ctp.response.casesvc.domain.repository.CaseIacAuditRepository;
+import uk.gov.ons.ctp.response.casesvc.representation.CategoryDTO;
 
-/**
- * The Case Iac Audit Service interface defines all business behaviours for operations on the Case
- * audit entities model.
- */
-public interface CaseIACService extends CTPService {
+@Service
+public class CaseIACService {
 
-  String generateNewCaseIACCode(final Integer casePk);
+  private static final Logger log = LoggerFactory.getLogger(CaseIACService.class);
 
-  String findCaseIacByCasePK(int caseFK);
+  private final InternetAccessCodeSvcClient iacClient;
+  private final CaseIacAuditRepository caseIacAuditRepo;
+  private final CaseEventRepository caseEventRepo;
+  private final Clock clock;
 
-  CaseIacAudit findCaseByIac(String iac) throws CTPException;
+  @Autowired
+  public CaseIACService(
+      final InternetAccessCodeSvcClient iacClient,
+      final CaseIacAuditRepository caseIacAuditRepo,
+      final CaseEventRepository caseEventRepo,
+      final Clock clock) {
+    this.iacClient = iacClient;
+    this.caseIacAuditRepo = caseIacAuditRepo;
+    this.caseEventRepo = caseEventRepo;
+    this.clock = clock;
+  }
+
+  public String findCaseIacByCasePK(int caseFK) {
+    log.debug("Entering findCaseIacByCasePK");
+
+    CaseIacAudit caseIacAudit = caseIacAuditRepo.findTop1ByCaseFKOrderByCreatedDateTimeDesc(caseFK);
+
+    if (caseIacAudit == null) {
+      return null;
```

Comment:
I notice that we behave differently in `findCaseByIac` i.e. we throw an exception, which is my preferred way of handling an illegal state